components/ResourceDetail/index.vue

```diff
@@ -251,10 +236,18 @@ export default {
       return null;
     },
   },
+
   watch: {
     asYamlInit(neu) {
       this.asYaml = neu;
     }
+  },
+
+  methods: {
+    // reading yamls from files is most easily tracked when done down in the component that handles other yaml-editing input, YamlEditor, but visually the button to upload lives up here
+    readFromFile() {
+      this.$refs.resourceYaml.$refs.yamlEditor.$refs.yamluploader.click();
```

Comment:
`$refs` should only be used inside the component they're in, nobody will expect or go looking for random other references in the entire rest of the codebase when editing that component.  Expose a method on the component(s) to trigger this instead.